src/Configuration/Properties/Property.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelCom\Integration\Configuration\Properties;
+
+class Property
+{
+    public function __construct(
+        private readonly string $name,
+        private readonly PropertyType $type,
+        private readonly ?string $description = null,
+        private readonly array $enum = [],
+        private readonly array $meta = [],
```

Comment:
```suggestion
        public readonly string $name,
        public readonly PropertyType $type,
        public readonly string $description,
        public readonly bool $isRequired = false,
        public readonly array $enum = [],
        public readonly array $meta = [],
```
I suggest we have `isRequired` for controlling which field is which is not. In addition, I suggest we have `description` mandatory. It'd be ugly to have fields without descriptions (we use them for labels).

I also suggest we make the properties public. The motivational driver for that is that then these object properties can be accessed for further processing by the particular marketplace. This is an interesting conundrum that manifests polar opposite approaches of coding that both make sense depending on circumstantial factors. 

### Properties in local code
Unless you have a good reason, properties should always revert to the most restricted variant which in `private`. That is because publicizing properties removes the boundary of where the data can go in the code and cause bugs due to unintended usage. Therefore `private` is an effective way of putting control over state and disallowing unintended usage. 

### _However..._
The code we have here becomes immutable once it reaches a particular marketplace. We cannot edit the accessors, nor we can add a method that processes the data inside _Property_ class because it's not owned by the particular marketplace. 
Hence, the rules change and `public` becomes applicable.